### ISSUE TITLE
Investigate cause(s) of front-end errors reported in Datadog 

### ISSUE THREAD

**Issue by fmccaf1** [department-of-veterans-affairs/va.gov-team#86854]
### User story

As a Cartography team member, I want to understand why Datadog is frequently capturing front-end errors for the landing page app, so that we can reduce the noise to reveal actual signal.

### Notes
- [per Patrick](https://dsva.slack.com/archives/C07E7JN4YPJ/p1721928292382549) these errors are known not to impact Veterans directly. However, they generate noise that makes it harder to notice actual issues.
- [DD front-end error report](https://vagov.ddog-gov.com/rum/error-tracking?query=%40application.id%3A1f81f762-c3fc-48c1-89d5-09d9236e340d&fromUser=true&source=all&from_ts=1721841726641&to_ts=1721928126641&live=true)
- Possible KPI: [Sessions with Errors](https://vagov.ddog-gov.com/rum/performance-monitoring?query=%40application.id%3A1f81f762-c3fc-48c1-89d5-09d9236e340d%20%40session.type%3Auser&filters=%5B%22env%22%2C%22service%22%2C%22version%22%2C%22%40session.type%22%2C%22%40geo.country%22%2C%22%40device.type%22%2C%22%40usr.email%22%5D&fromUser=false&tab=overview&from_ts=1721331174976&to_ts=1721935974976&live=true) – which is at 33% (for the past week) as of july 26
- Possible resolution paths include:
  - Fix
  - Create PR or ticket for team that owns error
  - Figure out how to "mute" specific errors in the DD report

Possible tasks:

## Acceptance criteria
- Documented in this ticket what's causing errors
- Reduced impact of errors in DD report (see possibilities above), as possible

**Comment by dcloud** [comment]
Related: [Sentry filters for "noisy" errors](https://github.com/department-of-veterans-affairs/va.gov-team-sensitive/blob/master/teams/benefits-decision-reviews/monitoring/sentry-filters.md)

**Comment by dcloud** [comment]
There's a category of errors around [Content-Security Policy (CSP)](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy), which is a set of directives from the server about types of resources the browser is allowed to load. Given these appear across apps, it seems probable that these result from one or more vendor packages, though it's unclear which ones may be the source.

You can filter to CSP errors in DDG error tracking by searching/filtering by `@error.message:*csp_violation*`. 

The top CSP-related error seen across VA.gov is `'blob' blocked by 'worker-src' directive` (`@error.message:*worker-src*`), which refers to errors where scripts are trying to launch [web workers](https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API/Using_web_workers) (scripts that can run in the background). The CSP for VA.gov includes `worker-src 'self' blob:;`, which appears to be a directive to allow 'blob' workers, so it's not clear why this is an error. 

The CSP returned by the server is:

```
default-src 'self'; base-uri http://*.vetsgov-internal https://*.va.gov https://optimize.google.com; block-all-mixed-content ; child-src https://*.uservoice.com https://www.googletagmanager.com https://www.youtube.com https://optimize.google.com https://resource.digital.voice.va.gov https://nebula-cdn.kampyle.com blob:; connect-src 'self' http://localhost:4000 https://*.va.gov https://api.mapbox.com https://events.mapbox.com https://www.google-analytics.com https://stats.g.doubleclick.net http://*.vetsgov-internal https://idp.int.identitysandbox.gov https://staging-va-gov-assets.s3-us-gov-west-1.amazonaws.com https://staging-va-gov-maintenance-windows.s3-us-gov-west-1.amazonaws.com https://s3-us-gov-west-1.amazonaws.com https://analytics.foresee.com https://brain.foresee.com https://survey.foreseeresults.com https://device.4seeresults.com https://health.foresee.com https://gateway.foresee.com https://resource.digital.voice.va.gov https://raw.githubusercontent.com wss://northamerica.directline.botframework.com https://northamerica.directline.botframework.com https://*.speech.microsoft.com wss://*.speech.microsoft.com https://search.usa.gov https://udc-neb.kampyle.com https://rum.browser-intake-ddog-gov.com https://session-replay.browser-intake-ddog-gov.com; font-src 'self' data: https://fonts.gstatic.com https://gateway.foresee.com https://medallia.s3.amazonaws.com https://staging-va-gov-assets.s3-us-gov-west-1.amazonaws.com https://s3-us-gov-west-1.amazonaws.com; form-action 'self' https://vicbdc.vba.va.gov https://vicbdc.ppd.va.gov https://resource.digital.voice.va.gov https://public.govdelivery.com; frame-ancestors 'none'; frame-src https://dap.digitalgov.gov https://resource.digital.voice.va.gov https://www.googletagmanager.com https://staging-va-gov-assets.s3-us-gov-west-1.amazonaws.com https://www.youtube.com; img-src 'self' data: blob: https://*.gstatic.com https://api.mapbox.com https://www.google-analytics.com https://www.googletagmanager.com https://*.va.gov https://optimize.google.com https://gateway.foresee.com https://static.foresee.com https://udc-neb.kampyle.com https://staging-va-gov-assets.s3-us-gov-west-1.amazonaws.com https://s3-us-gov-west-1.amazonaws.com https://ok6static.oktacdn.com https://dvp-oauth-application-directory-logos.s3-us-gov-west-1.amazonaws.com https://i.ytimg.com; manifest-src 'none'; media-src 'none'; object-src 'self' blob:; script-src 'self' 'unsafe-inline' 'nonce-HiF0x8sSeGasU85H66fmYuuRa4eVk8ka' http://www.google-analytics.com https://*.uservoice.com https://dap.digitalgov.gov https://designsystem.digital.gov https://maps.googleapis.com https://standards.usa.gov https://www.google-analytics.com https://www.googletagmanager.com https://tagmanager.google.com 'unsafe-eval' https://optimize.google.com https://gateway.foresee.com https://resource.digital.voice.va.gov https://nebula-cdn.kampyle.com https://staging-va-gov-assets.s3-us-gov-west-1.amazonaws.com https://s3-us-gov-west-1.amazonaws.com https://dsva-vetsgov-scorecard-staging.s3.us-gov-west-1.amazonaws.com https://staging.va.gov https://www.youtube.com https://*.ytimg.com https://cdn.botframework.com 'strict-dynamic'; style-src 'self' 'unsafe-inline' https://fonts.googleapis.com https://tagmanager.google.com https://optimize.google.com https://fonts.googleapis.com https://gateway.foresee.com https://medallia.s3.amazonaws.com https://staging-va-gov-assets.s3-us-gov-west-1.amazonaws.com https://s3-us-gov-west-1.amazonaws.com; worker-src 'self' blob:;  report-uri /csp-report
```

**Comment by dcloud** [comment]
The `TypeError: undefined is not an object (evaluating 'a.isProxied')` is most likely coming from the [third-party stencil library](https://github.com/ionic-team/stencil/issues/5035) that the web components use, perhaps due to scenarios where the app code attempts to use a web component before the web-components.entry.js is loaded. This is a guess based on my read of the stencil issue.

Potential fixes would involve ensuring the web-components bundle is loaded before React apps attempt to render web components. 

Again, Sentry ignores these errors.

**Comment by dcloud** [comment]
There are errors logged for things like browser extensions such as password managers. It's easy to forget that browser extensions inject CSS and JavaScript into web pages, or attempt to. Here are a few:

- [Trend Micro Password Manager cannot inject CSS](https://vagov.ddog-gov.com/rum/error-tracking/issue/a3bb5f0a-4614-11ef-8b2e-da7ad0900007?from_ts=1721834350050&to_ts=1723043950050&live=true)
- [Paypal Honey extension cannot inject an image](https://vagov.ddog-gov.com/rum/error-tracking/issue/93df2568-4519-11ef-a978-da7ad0900007?fromUser=false&refresh_mode=sliding&view=spans&from_ts=1721834563035&to_ts=1723044150342&live=true)
- [A chrome extension unable to inject a font](https://vagov.ddog-gov.com/rum/error-tracking/issue/1ff5c16e-4acf-11ef-aa9f-da7ad0900007?from_ts=1721834751802&to_ts=1723044351801&live=true)

**Comment by dcloud** [comment]
Essentially, there are many errors that arise due to the content-security policy on VA.gov, but probably for good reason.

**Comment by dcloud** [comment]
I talked with Robin from the benefits team and he had gone through the same exercise of compiling errors and sharing with the platform. Unclear if there are tickets that I can't see related to this issues in their backlog

- https://github.com/department-of-veterans-affairs/va.gov-team/issues/66883
- https://dsva.slack.com/archives/C04KW0B46N5/p1722989254861839
- https://dsva.slack.com/archives/CBU0KDSB1/p1700448514126239

Here is a ticket related to errors that had been seen:

- https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/1164


Can't find one for the `isProxied` error that he reported

**Comment by dcloud** [comment]
I think in terms of work, this is done, but possible next steps are with OCTO and the platform given some errors come from security policies, how FE apps depend on external bundle VADS components, and errors from libraries, some managed by the platform

**Comment by dcloud** [comment]
Per Wes, he is interested in what's on the dashboard. This [Errors, non-CSP, count

](https://vagov.ddog-gov.com/dashboard/e4a-257-6wa/mhv-on-vagov?fromUser=false&fullscreen_end_ts=1723660465779&fullscreen_paused=false&fullscreen_refresh_mode=sliding&fullscreen_section=edit&fullscreen_start_ts=1723055665779&fullscreen_widget=1236872126377774&refresh_mode=sliding&view=spans&from_ts=1723055433248&to_ts=1723660233248&live=true) indicates a large number of logged `SyntaxError` messages, which appear as a small number the [Error tracker](https://vagov.ddog-gov.com/rum/error-tracking?query=%40application.id%3A1f81f762-c3fc-48c1-89d5-09d9236e340d%20%40error.type%3ASyntaxError&et-side=activity&fromUser=false&issue_states=open&order=total_count&refresh_mode=sliding&source=all&view=spans&from_ts=1722451157504&to_ts=1723660757504&live=true)

**Comment by dcloud** [comment]
@wesrowe Given the ticket only mention the [Datadog error report](https://vagov.ddog-gov.com/rum/error-tracking?query=%40application.id%3A1f81f762-c3fc-48c1-89d5-09d9236e340d%20%40application.name%3A%22MHV%20on%20VA.gov%22%20env%3Aproduction&et-side=activity&fromUser=false&issue_states=open&order=total_count&refresh_mode=sliding&source=all&view=spans&from_ts=1722451157504&to_ts=1723660757504&live=true) and not the dashboard (which shows errors that don't appear in the error-tracking report, do you want another ticket to update the dashboard? I'm not sure why the dashboard has error things that don't appear in the error tracker!

**Comment by dcloud** [comment]
Looking at the [dashboard](https://vagov.ddog-gov.com/dashboard/e4a-257-6wa/mhv-on-vagov?fromUser=false&refresh_mode=sliding&view=spans&from_ts=1723055433248&to_ts=1723660233248&live=true), [one session](https://vagov.ddog-gov.com/rum/sessions?query=%40application.id%3A1f81f762-c3fc-48c1-89d5-09d9236e340d%20env%3Aproduction%20%40error.type%3ASyntaxError%20%40type%3Aerror&agg_m=count&agg_m_source=base&agg_t=count&cols=has_replay%2Cstatus%2Ctimestamp%2C%40view.name%2C%40session.id%2C%40error.source%2C%40error.message%2C%40error.is_crash&event=AgAAAZEzR83Dxa0YOQAAAAAAAAAYAAAAAEFaRXpSODNEQUFCc1FTN0ZnSmdpZnBVOQAAACQAAAAAMDE5MTMzOTMtYTRjOS00ZDQ5LTkyNDAtZDU2YjM3MmFjZDVj&fromUser=false&p_tab=session&refresh_mode=paused&sort_by=timestamp&sort_order=desc&track=rum&type=rum&viz=stream&from_ts=1723140000000&to_ts=1723147200000&live=false) is responsible 2274 errors that started with a blocked call to a domain that appears to be associated with an adware/malware browser extension. 2273 errors appear after the blocked domain, which could be subsequent errors from the malicious extension.

A [different session](https://vagov.ddog-gov.com/rum/sessions?query=%40type%3Aerror%20%40application.name%3A%22MHV%20on%20VA.gov%22%20env%3Aproduction%20-%40browser.name%3AHeadlessChrome%20-%40error.message%3A%2Acsp_violation%2A%20%40error.type%3ASyntaxError&agg_m=count&agg_m_source=base&agg_t=count&cols=has_replay%2Cstatus%2Ctimestamp%2C%40view.name%2C%40usr.id%2C%40session.id%2C%40error.source%2C%40error.message%2C%40error.is_crash&event=AgAAAZE4SrlB1z9gwgAAAAAAAAAYAAAAAEFaRTRTcmxCQUFCaEcxcWNfMzBrdzdKRwAAACQAAAAAMDE5MTM4YmEtMDFjOC00YTIzLWIzODEtMzA1NWEzY2Y4Nzcx&fromUser=false&refresh_mode=paused&sort_by=timestamp&sort_order=desc&track=rum&type=rum&viz=stream&from_ts=1723204800000&to_ts=1723248000000&live=false) responsible for 628 errors shows a similar pattern for a attempting to make a request to a not-allowed domain, then generating a series of `SyntaxError` messages.
